core/src/main/java/io/questdb/griffin/engine/functions/json/JsonPathFunctionFactory.java

```diff
@@ -0,0 +1,175 @@
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.json;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.CairoException;
+import io.questdb.cairo.ColumnType;
+import io.questdb.cairo.sql.Function;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.std.IntList;
+import io.questdb.std.ObjList;
+import io.questdb.std.str.DirectUtf8Sink;
+import org.jetbrains.annotations.NotNull;
+
+public class JsonPathFunctionFactory implements FunctionFactory {
+    private static final String FUNCTION_NAME = "json_path";
+    private static final String SIGNATURE = FUNCTION_NAME + "(ØøV)";
+
+    @Override
+    public String getSignature() {
+        return SIGNATURE;
+    }
+
+    @Override
+    public Function newInstance(
+            int position, ObjList<Function> args, IntList argPositions,
+            CairoConfiguration configuration, SqlExecutionContext sqlExecutionContext
+    ) {
+        final Function json = args.getQuick(0);
+        final Function path = args.getQuick(1);
+        final int targetType = parseTargetType(args.getQuiet(2));
+        final int onError = parseOnError(args.getQuiet(3));
+        final Function defaultValueFn = args.getQuiet(4);
+        final DirectUtf8Sink pointer = SupportingState.varcharConstantToJsonPointer(path);
+        final int maxSize = configuration.getStrFunctionMaxBufferLength();
+        final boolean strict = onError == JsonPathFunc.FAIL_ON_ERROR;
+        final JsonPathFunc fn = buildFunction(json, path, targetType, pointer, maxSize, strict);
+        applyDefault(fn, targetType, onError, defaultValueFn);
+        return fn;
+    }
+
+    private static @NotNull JsonPathFunc buildFunction(
+            Function json,
+            Function path,
+            int targetType,
+            DirectUtf8Sink pointer,
+            int maxSize,
+            boolean strict) {
+        switch (targetType) {
+            case ColumnType.LONG:
+            case ColumnType.DOUBLE:
+            case ColumnType.FLOAT:
+            case ColumnType.BOOLEAN:
+            case ColumnType.SHORT:
+            case ColumnType.INT:
+                return new JsonConstPathPrimitiveFunc(targetType, json, path, pointer, strict);
+            case ColumnType.VARCHAR:
+                return new JsonPathVarcharFunc(json, path, pointer, maxSize, false);
+            default:
+                throw new UnsupportedOperationException("nyi");  // TODO: complete remaining types
```

Comment:
this should be SqlException, correctly positioned for the user (use target type position in the original SQL)